tutorials/18_diagnostics_lc2st.ipynb

```diff
@@ -0,0 +1,821 @@
+    "\n",
+    "label = \"Probabilities (class 0)\"\n",
+    "# label = r\"$\\hat{p}(\\Theta\\sim q_{\\phi}(\\theta \\mid x_0) \\mid x_0)$\"\n",
+    "\n",
+    "fig, axes = plt.subplots(len(thetas_star), 3, figsize=(9,6), constrained_layout=True)\n",
+    "for i in range(len(thetas_star)):\n",
+    "    probs_data, _ = lc2st.get_scores_on_observed_data(\n",
+    "        theta_o=post_samples_star[i][:1000],\n",
+    "        x_o=xs_star[i],\n",
+    "        return_probs=True\n",
+    "    )\n",
+    "    dict_probs_marginals = get_probs_per_marginal(\n",
+    "        probs_data[0],\n",
+    "        post_samples_star[i][:1000].numpy()\n",
+    "    )\n",
+    "    # 2d histogram\n",
+    "    marginal_plot_with_probs_intensity(\n",
+    "        dict_probs_marginals['0_1'],\n",
+    "        marginal_dim=2,\n",
+    "        ax=axes[i][0],\n",
+    "        n_bins=50,\n",
+    "        label=label\n",
+    "    )\n",
+    "    axes[i][0].scatter(\n",
+    "        ref_samples_star[i][:,0],\n",
+    "        ref_samples_star[i][:,1],\n",
+    "        alpha=0.2,\n",
+    "        color=\"gray\",\n",
+    "        label=\"True posterior\"\n",
+    "    )\n",
+    "\n",
+    "    # marginal 1\n",
+    "    marginal_plot_with_probs_intensity(\n",
+    "        dict_probs_marginals['0'],\n",
+    "        marginal_dim=1,\n",
+    "        ax=axes[i][1],\n",
+    "        n_bins=50,\n",
+    "        label=label,\n",
+    "    )\n",
+    "    axes[i][1].hist(\n",
+    "        ref_samples_star[i][:,0],\n",
+    "        density=True,\n",
+    "        bins=10,\n",
+    "        alpha=0.5,\n",
+    "        label=\"True Posterior\",\n",
+    "        color=\"gray\"\n",
+    "    )\n",
+    "\n",
+    "    # marginal 2\n",
+    "    marginal_plot_with_probs_intensity(\n",
+    "        dict_probs_marginals['1'],\n",
+    "        marginal_dim=1,\n",
+    "        ax=axes[i][2],\n",
+    "        n_bins=50,\n",
+    "        label=label,\n",
+    "    )\n",
+    "    axes[i][2].hist(\n",
+    "        ref_samples_star[i][:,1],\n",
+    "        density=True,\n",
+    "        bins=10,\n",
+    "        alpha=0.5,\n",
+    "        label=\"True posterior\",\n",
+    "        color=\"gray\"\n",
+    "    )\n",
+    "\n",
+    "    axes[i][1].set_title(\"marginal  1\")\n",
+    "    axes[i][2].set_title(\"marginal 2 \")\n",
+    "\n",
+    "    for j in range(3):\n",
+    "        axes[j][0].set_ylabel(f\"obsevation {j + 1}\")\n",
+    "    axes[0][2].legend()\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## The case of Normalizing Flows ($\\ell$-C2ST-NF)\n",
+    "\n",
+    "$\\ell$-C2ST can also be specialized for normalizing flows,leading to improved test performance. The idea is to train and evaluate the classifiers in the space of the base distribution of the normalizing flow, instead of the parameter space that can be highly structures. \n",
```

Comment:
```suggestion
    "$\\ell$-C2ST can also be specialized for normalizing flows, leading to improved test performance. The idea is to train and evaluate the classifiers in the space of the base distribution of the normalizing flow, instead of the parameter space that can be highly structured. \n",
```